src/components/Site/SiteEquipmentContent.tsx

```diff
@@ -29,7 +88,77 @@ export function SiteEquipmentContent({
                     Add or select a piece of equipment.
                 </div>
             )}
-            {/*TODO display equipment details here*/}
+            <Button className="addLogger" onClick={handleAddLoggerClick}>
+                Add Logger
+            </Button>
+            {selectorCollapsed ? null : (
+                <LoggerSelector siteId={siteId} unitName={unit?.name || ''} />
+            )}
+            {<TabView>{loggerTabs}</TabView>}
+        </div>
+    );
+}
+
+function LoggerTab(logger: LoggerObject): ReactElement {
+    const columns: GridColDef[] = [
+        { field: 'timestamp', headerName: 'timestamp', flex: 1 },
+    ];
+
+    let rows: any[] = [];
+
+    logger.data.forEach((dataPoint, index) => {
+        rows.push({
+            id: index,
+            timestamp: dataPoint.timestamp,
+        });
+    });
+
+    return (
+        <div className="loggerTab">
+            <h1>Logger Data</h1>
+            <DataGrid className="dataGrid" rows={rows} columns={columns} />
+        </div>
+    );
+}
+
+interface LoggerSelectorProps {
+    //ID of the site the equipment is a part of
+    siteId: string;
+
+    // The name of the equipment
+    unitName: string;
+}
+
+function LoggerSelector({ siteId, unitName }: LoggerSelectorProps) {
+    const loggers: LoggerCollection = useSelector(
+        (state: RootState) => state.loggers
+    );
+
+    var loggerCard: any = [];
+
+    for (const [id, loggerData] of Object.entries(loggers)) {
```

Comment:
This shouldn't display loggers that are already attached to a piece of equipment. We may need to add a field in the logger document to denote whether its attached to a piece of equipment or not.